iroh-relay/src/relay_map.rs

```diff
@@ -112,14 +112,37 @@ impl RelayMap {
         self.relays.write().expect("poisoned").remove(url)
     }
 
-    /// Extends this `RelayMap` with another one.
-    pub fn extend(&self, other: &RelayMap) {
+    /// Joins this `RelayMap` with another one into a new one
+    pub fn join(self, other: RelayMap) -> RelayMap {
```

Comment:
Isn't this called `chain` usually?  I think this is very similar to the `Iterator::chain` signature.